commerce/bookstore/python/load_data.py

```diff
@@ -0,0 +1,142 @@
+                    dest="verbose", action='store_true')
+args = vars(parser.parse_args())
+
+if args["verbose"]:  # if the argument was set
+    print("High verbosity option turned on.")
+    debug = True  # Shows verbose debug messages in the console output
+else:
+    debug = False  # No debug messages
+
+
+def parse_data_to_dictionaries(input):  # input.file as string: the path to the data file
+    if debug: print("Parsing of " + input["file"] + "started.")
+    items = []  # prepare empty list
+    with open(input("").file, encoding="UTF-8") as data:  # reads the input.file through a stream
+        for row in csv.DictReader(data, delimiter=";", skipinitialspace=True):  # iterate through rows
+            item = {key: value for key, value in row.items()}  # Creates an item. Keys are taken from the first row
+            items.append(item)  # adds the dictionary to the list of items
+    if debug: print("Parsing of " + input["file"] + " successful.")
+    return items  # items as list of dictionaries: each item representing a data item from the file at input.file
+
+
+def load_data_into_typedb(input, session):  # Requests generation of insert queries and sends queries to the TypeDB
+    """
+      :param input as class: has load method to build insert query. Object initiated with an item to insert
+      :param session: an established connection to the TypeDB off of which a transaction will be created
+    """
+    items = parse_data_to_dictionaries(input)  # parses csv file (input.file) to create a list of dictionaries
+    skip_count = 0  # counter of non-successful insert attempts
+    for item in items:  # for each item dictionary in the list (former row in csv file)
+        with session.transaction(TransactionType.WRITE) as transaction:  # Open transaction to write with session provided
+            input_object = input(item)  # This is an object of input class initiated with an item as a parameter
+            typeql_insert_query = input_object.load()  # This builds the corresponding TypeQL insert query from item
+            if typeql_insert_query != "":
+                if debug: print("Executing TypeQL Query: " + typeql_insert_query)
+                transaction.query().insert(typeql_insert_query)  # runs the query
+                transaction.commit()  # commits the transaction
+                # todo: Add a transaction result check. Increase skip_cont if nothing was inserted
+            else:
+                if debug: print("Item parsing resulted in empty query statement. Skipping this item —", item)
+                skip_count += 1
+    print("Inserted " + str(len(items) - skip_count) + " out of " + str(len(items)) + " items from [ "
+          + input_object.file + "] into TypeDB with", input.__name__)
+    return  # END of load_data_into_typedb()
+
+
+def load_data():  # Main data load function
+    with TypeDB.core_client("localhost:1729") as client:  # Establishing connection
```

Comment:
Let's store `"localhost:1729"` in `config.py`. This is also good practice for readers particularly as their project matures and connects to an external host.

Reply:
Fixed in https://github.com/vaticle/typedb-examples/pull/143/commits/49cbdd105f6e61afc309f774f1041b2e42dc5663